### PR TITLE
🆗 Show alert after dev sentry button is pressed

### PR DESCRIPTION
**Pull request by drewvolz**
A confirmation because feedback is nice.

### CODE REVIEW

File: source/views/settings/screens/overview/developer.js [hunk 1]
```diff
@@ -14,9 +15,11 @@ export class DeveloperSection extends React.Component<Props> {
 		Sentry.captureMessage('A Sentry Message', {
 			level: SentrySeverity.Info,
 		})
+		Alert.alert('Sentry message sent.')
```
Comment (hawkrives) [inline source/views/settings/screens/overview/developer.js]:
```suggestion
		Alert.alert('Sentry message sent')
```

File: source/views/settings/screens/overview/developer.js [hunk 2]
```diff
@@ -14,9 +15,11 @@ export class DeveloperSection extends React.Component<Props> {
...
 	}
 	sendSentryException = () => {
 		Sentry.captureException(new Error('Debug Exception'))
+		Alert.alert('Sentry exception sent.')
```
Comment (hawkrives) [inline source/views/settings/screens/overview/developer.js]:
```suggestion
		Alert.alert('Sentry exception sent')
```

**Review by hawkrives — DISMISSED** [review]
iOS, at least, doesn't do full sentences in alert titles

**Comment by rye** [comment]
I like this idea.

Another idea: perhaps also add a hook for `isDevMode` only:

```js
// set a callback after an event was successfully sent
// its only guaranteed that this event contains `event_id` & `level`
Sentry.setEventSentSuccessfully((event) => {
  Alert.alert(`Sent an event to Sentry: ${event.event_id}`)
});
```

File: source/views/settings/screens/overview/developer.js (hunk 1 above)
Comment (drewvolz) [reply source/views/settings/screens/overview/developer.js]: > HIG: If the title is a complete sentence, use sentence-style capitalization and appropriate ending punctuation. If the title is a sentence fragment, use title-style capitalization and don’t add ending punctuation.

...alright. I'll accept it.

File: source/views/settings/screens/overview/developer.js (hunk 2 above)
Comment (drewvolz) [reply source/views/settings/screens/overview/developer.js]: Same as message above.

**Comment by drewvolz** [comment]
@rye I actually looked for that bit of docs -- thank you for finding it!

**Review by rye — APPROVED** [review]
Looks alright.

**Comment by drewvolz** [comment]
I am unable to get this hook to work. If anyone can advise, I would love to add it.

**Comment by rye** [comment]
What is "this hook"? Do you mean `Sentry.setEventSentSuccessfully`?

**Comment by drewvolz** [comment]
Correct, for `Sentry.setEventSentSuccessfully` and maybe `Sentry.setShouldSendCallback`

**Comment by rye** [comment]
Bear in mind that `Sentry` will not be installed (read: activated) in local development environments unless you remove the `if (IS_PRODUCTION)`

**Comment by rye** [comment]
Actually, my suggestion is illogical.

Sentry only boots when `IS_PRODUCTION`, but we only want an alert for local testing, I think, which is `!IS_PRODUCTION`… and those have no overlap.

**Review by rye — APPROVED** [review]
Green light from me.

**Comment by hawkrives** [comment]
But if Sentry is not active, those buttons won't do anything at all, will they? They won't actually send the event?

**Comment by rye** [comment]
@hawkrives correct. Pressing the button will fire the callback, but nothing will happen internally.

**Comment by hawkrives** [comment]
I would argue for using the Sentry callbacks here, so that we know when the event has actually been sent. 

I'm fine with using an Alert if we're !IS_PRODUCTION, to announce that no alert will be sent, but I'd like the success notification in betas and nightlies.

**Comment by rye** [comment]
@hawkrives, if `!IS_PRODUCTION`, no `Sentry` callback will be set either, though, because `bootSentry()` will not install the `config`.

**Review by hawkrives — DISMISSED** [review]
Callback in production, but alert that no event was sent in development please.

**Comment by drewvolz** [comment]
@hawkrives that doesn't make sense to me.

**Comment by hawkrives** [comment]
Two cases:

1. **In development**: No data will ever be sent, so we should tell the user that nothing will appear in Sentry's dashboard.

2. **In production**: Data will be sent, so it makes sense to alert the user _after_ the data has been sent, instead of onPress, so that they know when they can expect to see it.

**Comment by drewvolz** [comment]
> so it makes sense to alert the user _after_ the data has been sent, instead of onPress, so that they know when they can expect to see it.

but we don't show these features outside of development?

**Comment by hawkrives** [comment]
We show them on betas and nightlies

**Comment by drewvolz** [comment]
I think this last commit covers that

**Comment by hawkrives** [comment]
Yes, it should; GitHub did not show me that commit when I was responding to your previous comment.

**Review by hawkrives — APPROVED** [review]

**Pull request merged**